src/pages/index.js

```diff
@@ -83,25 +89,122 @@ function handlePicturePopup(name, link) {
-  createCard(data);
-  popupNewCard.reset();
-  newCardFormValidator.disableSubmitButton();
-});
+const popupDelete = new PopupConfirmation(popupConfirmDelete, handleDeletePicture);
+
+function handleDeleteButton(cardId, element) {
+  popupDelete.open(cardId, element);
+}
+
+function handleDeletePicture(cardId, element) {
+  api
+    .deleteCard(cardId)
+    .then(() => {
+      element.remove();
+    })
+    .catch((error) => {
+      console.error('Error deleting card:', error);
+    });
+}
+
+function handleLikeButton(cardId, isLiked, btnLike) {
+  if (isLiked) {
+    api
+      .dislikeCard(cardId)
+      .then(() => {
+        btnLike.classList.remove('card__btn-like_active');
+        this._isLiked = false;
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  } else {
+    api
+      .likeCard(cardId)
+      .then(() => {
+        btnLike.classList.add('card__btn-like_active');
+        this._isLiked = true;
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  }
+}
+
+const popupNewCard = new PopupWithForm(newCard, handleNewCardSubmition);
+
+function handleNewCardSubmition(cardData) {
+  api
+    .addCard(cardData)
+    .then(() => {
+      createCard(cardData);
+    })
+    .then(() => {
+      popupNewCard.reset();
+      newCardFormValidator.disableSubmitButton();
+    })
+    .catch((error) => {
+      console.error('Error adding a new card:', error);
+    });
+}
 
 btnNewCard.addEventListener('click', () => {
   popupNewCard.open();
 });
 
-const userInfo = new UserInfo({ nameSelector: '.profile__name', aboutMeSelector: '.profile__title' });
+const userInfo = new UserInfo({ nameSelector: '.profile__name', aboutSelector: '.profile__title', avatarSelector: '.profile-photo-edit__image' });
+
+api.getData().then(([userData, userCards]) => {
+  console.log(userData, userCards);
+  userInfo.setUserInfo(userData);
+  cardList = new Section(
+    {
+      data: userCards,
+      renderer: createCard,
+    },
+    containerSelector // .cards
+  );
+  cardList.renderItems();
+});
+
+const editProfileImage = new PopupWithForm(popupEditProfileImage, handleProfileImageSubmit);
 
-const popupEditProfile = new PopupWithForm(editProfile, (data) => {
-  userInfo.setUserInfo(data);
+btnEditProfileImage.addEventListener('click', () => {
+  editProfileImage.open();
 });
 
+function handleProfileImageSubmit(userData) {
+  api
+    .updateProfileImage(userData)
+    .then(() => {
+      userInfo.setUserAvatar(userData);
+    })
+    .then(() => {
+      editProfileImage.reset();
+    })
+    .catch((error) => {
+      console.error('Error updating a new image:', error);
+    });
+}
+
+const popupEditProfile = new PopupWithForm(editProfile, handleUserProfileSubmition);
+
+function handleUserProfileSubmition(userData) {
+  console.log(userData);
+  api
+    .updateUserInfo(userData)
+    .then(() => {
+      userInfo.setUserInfo(userData);
+    })
+    .then(() => {
```

Comment:
Needs correction: there is no need to have two `then` blocks, you can do everything via one

Reply:
Still needs correction